### ISSUE TITLE
Debian packaging for debian stretch

### ISSUE THREAD

**Issue by jgeysens** [aptly-dev/aptly#741]
## Detailed Description
The aptly debian package is out-of-date and that is problematic for debian stretch.

## Possible Implementation
Make sure the debian package installs correctly on debian stretch.  It should depend on the gnupg1 and gpgv1 packages for Debian stretch (breaking change in aptly version 1.3.0).

Even after installing both packages, keys that worked before will not longer work after the upgrade to 1.3.0 (since keys would typically be generated with the default gpg version in stretch - it worked in 1.2.0, so this is a regression)

## Context
`ERROR: unable to initialize GPG signer: looks like there are no keys in gpg, please create one (official manual: http://www.gnupg.org/gph/en/manual.html)`

## Your Environment
Debian Stretch + Aptly 1.3.0 (upgrade from 1.2.0)

**Comment by jgeysens** [comment]
Suggestion : having a package for stretch would solve the dependency issue.

> Second issue is interesting - so signing worked for you in 1.2.0 with default gpg (GnuPG 2.x), but as 1.3.0 falls back to gpg1, gpg1 can't read 2.x keyring?

Correct.  I currently reverted aptly to version 1.2.0 - I don't want to mess up our current configuration, and re-create the keys.

Linking issue #740

**Comment by apachelogger** [comment]
@smira WRT to the second issue, that is an expected side effect of restricting the version. We were semi-compatible with gpg2 iff the gpg implementation was used and iff the interaction extends no further than auth-less signing (e.g. passphrase-less key). Or perhaps a truer statement is: gpg2 CLI is semi-compatible with gpg1 CLI, so we were too :smile:. gpg1 however has no notion or understanding of gpg2 (it's a one-way street, gpg2 imports gpg1 keyrings once and then they go their sepearate ways, gpg1 doesn't do anything of the sort with gpg2). So, essentially what is happening is if you only ever used gpg2 and it happend to _just_ work as per the restriction I mentioned above, then upgrading to 1.3.0 and thus going back to gpg1 means you have no keys (because the keys are in gpg2's keyring and gpg1's will be empty). There isn't really much to be done about this from our side, except for actually building gpg2 support.

Now that I wrote that, perhaps we should put a note about this in the release notes, and tell people that they need to export their keys from gpg2 and import into gpg1 if they happen to have used only gpg2 before?

**Comment by apachelogger** [comment]
@smira You are right, this won't ever be win-win until gpg2 support materializes. Until then I'd argue that it's easier to explain that aptly "only supports gpg1" rather than "it supports gpg1, but gpg2 may work if you do xyz" though. But then, I also don't care much ^^

An env var seems a reasonable comprimise and adding an `APTLY_NO_GPG_VERSION` env check to the `cliVersionCheck` func should be taking care of that.

**Comment by apachelogger** [comment]
> which enables both gpg and openpgp aptly signers: gnupg.org/faq/whats-new-in-2.1.html#nosecring

That is what I was talking about documenting. If someone happened to successfully use gpg2 with aptly1.2, they could do an export-import dance from gpg2 to gpg1 and then the keys would once again be available to aptly.

**Comment by smira** [comment]
So, do we agree to add advice on gpg2 to the release notes & www.aptly.info as interim solution until we implement gpg2 support?

**Comment by jgeysens** [comment]
Exporting gpg2 keys + importing into gpg1 seems to work.

```
gpg --output gpg2_exported_pub.gpg --armor --export xxxxxxxxxxxxxxxxx
gpg --output gpg2_exported_sec.gpg --armor --export-secret-key xxxxxxxxxxxxxxxxx

gpg1 --import gpg2_exported_pub.gpg
gpg1 --import --allow-secret-key-import gpg2_exported_sec.gpg
```

**Comment by apachelogger** [comment]
@smira yep. the previous comment's instructions look reasonable if anyone wants to write up the change for the website

**Comment by absltkaos** [comment]
My main issue with using gpg1 is that on ubuntu (specifically xenial at the moment)... they did this silly thing, where gpg1 (package `gnupg`) is available. So is package `gnupg2`. However because I have a passphrase on my signing keys I need to have gpg agent running and ubuntu xenial _only_ has the gpg2 version of gnupg-agent, which is _not_ compatible with gpg1.

Now, that being said... the main thing holding me back from using the "internal" gpg support in 1.3.0 is that I don't know how to allow aptly to use my signing key, that requires a passphrase, in as secure a manner as using the gpg-agent. (I'd rather not store the passphrase on disk, in a config file)

**Comment by jspricke** [comment]
In addition to @jgeysens commands I had to do:
`
gpg1 --import --no-default-keyring  --keyring trustedkeys.gpg gpg2_exported_pub.gpg
`

**Comment by smira** [comment]
I'm working on a PR to introduce set of basic tests for internal PGP, GPG1 and GPG2 so that it's easy to see what is missing. Once I get them running I hope to establish compatibility with GnuPG 2.x

**Comment by smira** [comment]
Should be fixed with #779
